src/java/org/apache/cassandra/db/ConsistencyLevel.java

```diff
@@ -230,9 +230,10 @@ public void validateForWrite(String keyspaceName, QueryState queryState) throws
     }
 
     // This is the same than validateForWrite really, but we include a slightly different error message for SERIAL/LOCAL_SERIAL
-    public void validateForCasCommit(AbstractReplicationStrategy replicationStrategy, String keyspaceName, QueryState queryState) throws InvalidRequestException
+    public void validateForCasCommit(AbstractReplicationStrategy replicationStrategy, String keyspaceName, QueryState queryState, boolean skipGuardrail) throws InvalidRequestException
```

Comment:
I think guardrail is to protect users from shooting themself. The CL.ANY here is hard coded in code base, user should not configure guardrail to block it.

Reply:
My 2 cents: I agree it makes sense to allow to skip CL validation in this case because the CL is hard coded in the codebases (as noted by Zhao). I also agree it is not great by an API point of view, but `StorageProxy#doPaxos` is `private` so exposure is limited, and I'd rather be pragmatic here than rewriting things.